### PR TITLE
raftstore: use force_send to send ApplyRes

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <hustmh@gmail.com>

### What is changed and how it works?

Issue Number: Close #13160

What's Changed:

```commit-message
Use force_send to send ApplyRes
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more MEM

### Release note 

```release-note
Fix a bug that regions may be overlapped if raftstore is too busy
```

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]
LGTM

File: components/raftstore/src/store/fsm/store.rs [hunk 1]
```diff
@@ -292,16 +292,21 @@ where
...
-                r.region_id,
+            let region_id = r.region_id;
+            if let Err(e) = self.router.force_send(
+                region_id,
                 PeerMsg::ApplyRes {
                     res: ApplyTaskRes::Apply(r),
                 },
-            );
+            ) {
+                error!("failed to send apply result"; "region_id" => region_id, "err" => ?e);
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/store.rs:302]: What if it's shutting down?

Comment (5kbpers) [reply components/raftstore/src/store/fsm/store.rs:302]: Already checked here https://github.com/tikv/tikv/blob/1e13ddf3bf12c00afb4d049d05978fecae9a6067/components/batch-system/src/router.rs#L263

Comment (gengliqi) [inline components/raftstore/src/store/fsm/store.rs:302]: Seems the error log is printed only when the channel is closed. 
So is it necessary to add this error log?

**Review by gengliqi — APPROVED** [review]
LGTM

File: components/raftstore/src/store/fsm/store.rs (hunk 1 above)
Comment (5kbpers) [reply components/raftstore/src/store/fsm/store.rs:302]: A little is better than none. 🤣

**Pull request merged**